### PR TITLE
[IMP] base: streamlining command argparser, description, epilog

### PR DESCRIPTION
**Pull request by lordkrandel**
Refactored all Commands to have a `self.parser` without having to manually instantiate one.

Command descriptions shown on their single command's help improved:
- refactored to Command class
- used `inspect.cleandoc` to have a better output.

Commands' parsers are now all given `Command.epilog` which defaults to None but can be overridden by subclasses.

Reviewed some useless imports and license messages that got in the way.

Task [link](https://www.odoo.com/odoo/project/1364/tasks/4350863)
task-4350863

### CODE REVIEW

File: odoo/cli/command.py [hunk 1]
```diff
@@ -14,11 +15,27 @@
...
+    @property
+    def parser(self):
+        if not self._parser:
+            def cleanup(x):
+                return textwrap.dedent(x or "").strip()
+            self._parser = argparse.ArgumentParser(
+                formatter_class=argparse.RawDescriptionHelpFormatter,
+                prog=self.prog_name,
+                description=cleanup(self.description or self.__doc__),
+                epilog=cleanup(self.epilog),
```
Comment (Julien00859) [inline odoo/cli/command.py:35]: [`inspect.cleandoc`](https://docs.python.org/3/library/inspect.html#inspect.cleandoc)
```suggestion
                description=inspect.cleandoc(self.description or self.__doc__),
                epilog=inspect.cleandoc(self.epilog),
```

File: odoo/cli/help.py [hunk 2]
```diff
@@ -6,17 +7,17 @@
 class Help(Command):
     """ Display the list of available commands """
 
-    template = """\
-usage: {prog_name} [--addons-path=PATH,...] <command> [...]
+    template = textwrap.dedent("""
```
Comment (Julien00859) [inline odoo/cli/help.py:10]: dedent doesn't strip
```suggestion
    template = textwrap.dedent("""\
```

Comment (lordkrandel) [reply odoo/cli/help.py:10]: ... doesn't strip? 😉 😉 

immagine

Ok, it's terrible and inappropriate.
Thanks for the advice. 👍🏻

**Review by kmagusiak — COMMENTED** [review]
Code looks good, just have to try it.

File: odoo/cli/__init__.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
 # Import just the command, the rest will get imported as needed
-from .command import Command, main  # noqa: F401
-from .help import Help  # noqa: F401
+
+from .command import Command, commands  # noqa: F401
```
Comment (Julien00859) [inline odoo/cli/__init__.py:3]:
```suggestion
from .command import Command  # noqa: F401
```
why would we need to export `commands`? `help` can go and import it directly from `odoo.cli.command` doesn't it?

File: odoo/cli/loading.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import sys
```
[the remark this answers is not in the document]
Comment (lordkrandel) [reply odoo/cli/loading.py:1]: I agree... but namespace pollution blah blah... reverting.

**Review by Julien00859 — APPROVED** [review]
@robodoo r+

**Comment by Julien00859** [comment]
Hello @odoo/rd-security, this is once again a PR that moves stuff that are part of the CLI. So we have `Path(sys.argv[0]).name` to determine how the CLI was invoked to produce a nice "usage: " with each command's --help. We also use `inspect.cleandoc` which is a `textwrap.dedent` with extra stripping capabilities, I believe this is a wrong-positive. There's nothing new, just stuff that moved, was refactored.

**Comment by mart-e** [comment]
@robodoo override=ci/security

**Pull request closed without merging**